tests/test_transformations.py

```diff
@@ -0,0 +1,49 @@
+import mlx.core as mx
+from mlx_graphs.utils.transformations import (
+    to_edge_index,
+    to_sparse_adjacency_matrix,
+    check_adjacency_matrix,
+)
+import pytest
+
+
+@pytest.mark.parametrize(
+    "x, expected_exception",
+    [
+        (mx.array([[0, 1, 0], [1, 0, 1], [0, 1, 0]]), None),  # Valid matrix
+        (mx.array([[0, 1, 0], [1, 0, 1]]), ValueError),  # Non-square matrix
+        (mx.array([0, 1, 0, 1, 0, 1]), ValueError),  # 1D matrix
+        (mx.array([[[0]]]), ValueError),  # 3D matrix
+    ],
+)
+def test_check_adjacency_matrix(x, expected_exception):
+    @check_adjacency_matrix
+    def foo(adjacency_matrix):
+        return True
+
+    if expected_exception:
+        with pytest.raises(expected_exception):
+            foo(adjacency_matrix=x)
+    else:
+        assert foo(adjacency_matrix=x) is True
+
+
+def test_to_edge_index():
+    matrix = mx.array([[0, 1, 2], [3, 0, 0], [5, 1, 2]])
+    edge_index = to_edge_index(matrix)
+    assert edge_index.dtype == matrix.dtype
```

Comment:
To be consistent in all tests, we have to determine whether we use a descriptive message in the `assert` tests. What do you think about that?

Reply:
I personally used assert messages in the GAT PR as they provide some context in those tests, but I'm not sure it will be necessary in all tests.

Reply:
Yes, let's do that as much as we can.